src/main/java/com/snowflake/kafka/connector/internal/streaming/schemaevolution/iceberg/IcebergSchemaEvolutionService.java

```diff
@@ -61,4 +110,56 @@ public void evolveSchemaIfNeeded(SchemaEvolutionTargetItems targetItems, SinkRec
       }
     }
   }
+
+  /**
+   * Get only column names, ignore nested field names. Remove double quotes.
+   *
+   * <p>example: TEST_STRUCT.field1 -> TEST_STRUCT
+   */
+  private Set<String> getColumnsToEvolve(SchemaEvolutionTargetItems targetItems) {
+    return targetItems.getColumnsToAdd().stream()
+        // remove double quotes
+        .map(targetItem -> targetItem.split("\\.")[0].replaceAll("\"", ""))
```

Comment:
instead of commenting, you could just extract it to method removeDoubleQouetes :P